pkg/extension/extension.go

```diff
@@ -0,0 +1,22 @@
+package extension
+
+import (
+	"context"
+
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Extension interface{}
+
+type PreReconciliationExtension interface {
```

Comment:
The nomenclature of pre/post reconciliation is flawed, as extension runs become _part_ of reconciliation (and deletion is _also_ part of reconciliation). What is really meant here is that they run before/after the respective helm steps (pre/post upgrade or install, pre uninstall - why no post uninstall, though?)

Reply:
Agreed, the naming should be changed.

Reply:
I do agree that the naming could be misleading, because of what you say: the extensions are supposed to be part of reconciliation -- with that I mean, part of the `Reconcile()` method of the reconciler. But, the term is somewhat overloaded in the internals of the helm-operator-plugins. As I mentioned elsewhere, the underlying `ActionInterface` has a certain set of actions, corresponding to the different flows within the reconciler and this set of actions also includes a `Reconcile()` method, along with the methods which directly correspond to Helm actions.

I tend to believe that what we are aiming for could be Post/Pre wrappings for each of these methods, i.e. Post/PreInstall, Post/PreUninstall,, Post/PreUpgrade, Post/PreReconcile.

I agree that the overloading here is not ideal, but ignoring this for a second, what do you think of this?